### PR TITLE
[Storefront] Reformat apiClientService(Done)

### PR DESCRIPTION
**Pull request by HnKnA**
**- Stars:**
image
**- Ratings:**
image
**- Related products:**
image
**- Featured products:**
image
**- Product's detail:**
image

### CODE REVIEW

**Review by khanhtrand — APPROVED** [review]

File: storefront/modules/catalog/services/CategoryService.ts [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
 export async function getCategories(): Promise<Category[]> {
-  const response = await fetch('/api/product/storefront/categories');
+  const response = await apiClientService.get(baseUrl);
   return await response.json();
 }
 
 export async function getCategory(id: number) {
-  const response = await fetch('/api/product/storefront/categories/' + id);
+  const response = await apiClientService.get(baseUrl + id);
```
Comment (minhtridn2001) [inline storefront/modules/catalog/services/CategoryService.ts:12]: `apiClientService.get(baseUrl + id);`
Is the url missing the slash "/"?

File: storefront/modules/catalog/services/ProductService.ts [hunk 2]
```diff
@@ -4,53 +4,55 @@ import { ProductAll, ProductFeature } from '../models/ProductFeature';
...
     `${process.env.API_BASE_PATH}/product/storefront/product-option-combinations/${productId}/values`
   );
   if (res.status >= 200 && res.status < 300) return res.json();
-  return Promise.reject(res);
+  throw res;
 }
 
 export async function getProductByMultiParams(queryString: string): Promise<ProductAll> {
-  const res = await fetch(`/api/product/storefront/products?${queryString}`);
+  const res = await apiClientService.get(`/api/product/storefront/products?${queryString}`);
```
Comment (minhtridn2001) [inline storefront/modules/catalog/services/ProductService.ts:32]: Please be consistent in handling URLs:
#L18:
`process.env.API_BASE_PATH + '/product/storefront/product/' + slug`
#L32:
`/api/product/storefront/products?${queryString}`

File: storefront/modules/catalog/services/ProductService.ts [hunk 3]
```diff
@@ -4,53 +4,55 @@ import { ProductAll, ProductFeature } from '../models/ProductFeature';
...
 export async function getProductVariationsByParentId(
   parentId: number
 ): Promise<ProductVariation[]> {
-  const res = await fetch(
+  const res = await apiClientService.get(
     `${process.env.API_BASE_PATH}/product/storefront/product-variations/${parentId}`
   );
   if (res.status >= 200 && res.status < 300) return res.json();
-  return Promise.reject(res);
+  throw res;
```
Comment (minhtridn2001) [inline storefront/modules/catalog/services/ProductService.ts:43]: `res` is an instance of Response, so throwing the response is not appropriate.
Create an Error and throw it instead.

**Review by minhtridn2001 — CHANGES_REQUESTED** [review]
Hello @HnKnA,
I have a few comments. Please check it.
Thanks!

File: storefront/modules/catalog/services/CategoryService.ts (hunk 1 above)
Comment (HnKnA) [reply storefront/modules/catalog/services/CategoryService.ts]: Yes, I've configured the url in the newest code to resolve the problem

File: storefront/modules/catalog/services/ProductService.ts (hunk 2 above)
Comment (HnKnA) [reply storefront/modules/catalog/services/ProductService.ts]: I've reformatted and use #L32 as the common way to handle URLs

File: storefront/modules/catalog/services/ProductService.ts (hunk 3 above)
Comment (HnKnA) [reply storefront/modules/catalog/services/ProductService.ts]: I've created and returned the Error instead of throwing raw res

**Review by minhtridn2001 — APPROVED** [review]

**Pull request merged**